### PR TITLE
Non-API breaking introduction of Schnorr keys

### PR DESCRIPTION
**Pull request by dr-orlovsky**
First two steps of #588, which are non-API breaking and can get into `0.26.1`:

- Putting pre-Schnorr ECDSA keys under `util::ecdsa`, while still re-exporting them as `util::key` such that downstream dependencies will not break
- Updating all internal references to the keys to `util::ecdsa`
- Introducing `util::schnorr` mod re-exporting Secp256k1 Schnorr key types
- Making use of `util::key` ECDSA keys deprecated

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -85,9 +85,14 @@ pub use util::address::AddressType;
 pub use util::amount::Amount;
 pub use util::amount::Denomination;
 pub use util::amount::SignedAmount;
+pub use util::merkleblock::MerkleBlock;
+
+pub use util::ecdsa;
+pub use util::schnorr;
```
Comment (thomaseizinger) [inline src/lib.rs:91]: :+1: 

This is a tiny step towards https://github.com/rust-bitcoin/rust-bitcoin/pull/525 where the `util` module was made private.

**Review by thomaseizinger — COMMENTED** [review]
ACK 230813b57826ac8e4288f39848341a5669683ce4

**Review by apoelstra — APPROVED** [review]
ack 230813b57826ac8e4288f39848341a5669683ce4

**Comment by stevenroose** [comment]
utACK 230813b57826ac8e4288f39848341a5669683ce4 !

**Review by stevenroose — APPROVED** [review]

**Review by sgeisler — APPROVED** [review]
utACK 230813b57826ac8e4288f39848341a5669683ce4

**Pull request merged**